elements/centos7-vault/pre-install.d/01-set-centos-mirror

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+if [ ${DIB_DEBUG_TRACE:-0} -gt 0 ]; then
+    set -x
+fi
+set -eu
+set -o pipefail
+
+
+# Only set the mirror for the Base, Extras, Updates and CentOSPlus repositories
+# This assumes a fair amount about the structure of the CentOS-Base repo file.
+# Would be nice to use -E "s/^\[(base|updates|extras|centosplus)\]/\[\1_$DIB_CENTOS7_MIRROR_VERSION\]/g"
+
+sed -e "s|^#baseurl=http[s]*://mirror.centos.org/centos|baseurl=$DIB_CENTOS7_MIRROR|" \
```

Comment:
I guess the alternative is to include a current version of `CentOS-Vault.repo` and update it when there is a new release which you currently need to do anyway due to the switch on `DIB_CENTOS7_VERSION`.

Reply:
Is there a CentOS-Vault.repo file published on vault.centos.org?  If not I don't think there's a clear advantage that way.  This way we can also cope with (eg) internal mirrors or other CentOS repos.

Reply:
Not directly, but you can get it from the latest centos-release package, e.g: http://mirror.centos.org/centos/7/os/x86_64/Packages/centos-release-7-7.1908.0.el7.centos.x86_64.rpm. Like you say though, there probably isn't a massive advantage though.